From 5b3d72377789de4e3190705edb97fc402187c3d7 Mon Sep 17 00:00:00 2001
From: Garrett Brown <garrett.brown@aclima.io>
Date: Sat, 16 Nov 2019 14:05:50 -0800
Subject: [PATCH 2/2] Fix cross-compiling

This change fixes two errors:

----------------------------------------------------------------------------

file RPATH_CHANGE could not write new RPATH:

    /usr/lib

to the file:

    /bin/nodesize_dbg

No valid ELF RPATH or RUNPATH entry exists in the file;

----------------------------------------------------------------------------

/bin/sh: 1: nodesize_dbg: Exec format error
---
 CMakeLists.txt | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f373320..e7dd1ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,6 +37,16 @@ if(NOT foonathan_memory_FOUND)
       set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
     endif("${isSystemDir}" STREQUAL "-1")
   endif(BUILD_SHARED)
+
+  # Fix cross-compiling
+  list(APPEND extra_cmake_args -DCMAKE_SKIP_INSTALL_RPATH=ON)
+  if (CMAKE_SYSTEM_PROCESSOR STREQUAL amd64)
+    list(APPEND extra_cmake_args -DCMAKE_CROSSCOMPILING_EMULATOR=qemu-x86_64)
+  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL arm)
+    list(APPEND extra_cmake_args -DCMAKE_CROSSCOMPILING_EMULATOR=qemu-arm)
+  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL aarch64)
+    list(APPEND extra_cmake_args -DCMAKE_CROSSCOMPILING_EMULATOR=qemu-aarch64)
+  endif()
   
   if(DEFINED CMAKE_BUILD_TYPE)
     list(APPEND extra_cmake_args -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE})
-- 
2.20.1

